### PR TITLE
2961/layout fixes

### PR DESCRIPTION
**Pull request by tatiana-scandi**
Fixes 4 UI issues (I combined into one to avoid having 4 extra small pull requests):
https://github.com/scandipwa/scandipwa/issues/2961
https://github.com/scandipwa/scandipwa/issues/2959
https://github.com/scandipwa/scandipwa/issues/2975
https://github.com/scandipwa/scandipwa/issues/2974

Had to add another media queries set, as all the attempts to make it look not ugly with the existing ones failed.

### CODE REVIEW

**Comment by tatiana-scandi** [comment]
Closed, as recreating PR into another branch: https://github.com/scandipwa/scandipwa/pull/2983

File: packages/scandipwa/src/component/MyAccountOverlay/MyAccountOverlay.style.scss [hunk 1]
```diff
@@ -158,6 +158,13 @@
             padding-block-start: 14px;
             padding-inline: 14px;
         }
+
+        &_state_forgotPassword {
+            .Button.Button_likeLink {
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/MyAccountOverlay/MyAccountOverlay.style.scss:163]: Do we need to increase specificity here with two class names?

**Review by zans-laksa — CHANGES_REQUESTED** [review]

**Comment by Elena-Evsa** [comment]
#2961 - Passed 
#2959 - a. button 'Add to cart', compare and Wish list icons are moving good
b. Title and image should stay at one height - https://vimeo.com/578395827/6f1e9cc354, can we fix it?
c. In Product https://pr-2984.d25kd40decjpue.amplifyapp.com/default/song-of-ice-and-fire.html 'Sample' overlap the name of option then screen has been sized https://www.screencast.com/t/3sNccNUwvcan
d. Same product as in c. - last from option 'Bonus: Fire and Blood: 300 Years Before A Game of Thrones(+$0.00)' better place under the first word of the name of option, but not under the checkmark https://www.screencast.com/t/JyAPjQ77
#2975 - 'Sign in' bit shifted up https://www.screencast.com/t/l2L9hXe6 (same in Forgot password pop-up)
#2974 - posted comment in issue

File: packages/scandipwa/src/component/MyAccountOverlay/MyAccountOverlay.style.scss (hunk 1 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/component/MyAccountOverlay/MyAccountOverlay.style.scss:163]: Yes, you need.

But actually this is due to too general selector used in another part of code. I added more classes to elements I want to style + got rid of too general `.Button` selector => 2 class names are not needed any more.

**Comment by tatiana-scandi** [comment]
1. Title and image should stay at one height - FIXED
2. One/two column layout adjusted for shipping methods and payment methods
3. Long downloadable links position - FIXED. But next time, please, create a separate bug for a new issue found when testing another bugfix. (It has not been mentioned initially + it is not the issue caused by this PR breaking other functionality). 
4. 'Sign in' bit shifted up - FIXED

**Comment by Elena-Evsa** [comment]
#2961 - Passed
#2959 - a. Passed
b. Passed
c. and d. - please have a look at mobile version, fixed on desktop
#2975 - 'Sign in' bit shifted up - can reproduce [https://www.screencast.com/t/3NT6ejjqpjo](https://www.screencast.com/t/3NT6ejjqpjo)
#2974 - Passed

**Comment by tatiana-scandi** [comment]
@Elena-Evsa please recheck remaining fixes

**Comment by carinadues** [comment]
c. and d. have passed in mobile. 
#2975 - 'Sign in' isn't shifted up anymore.

File: packages/scandipwa/src/component/ProductDownloadableLinks/ProductDownloadableLinks.component.js [hunk 2]
```diff
@@ -98,15 +75,34 @@ export class ProductDownloadableLinks extends PureComponent {
...
+              isOpenInNewTab={ isOpenInNewTab }
+              block="ProductDownloadableLink"
+              elem="SampleLink"
+            >
+                { __('Sample') }
+            </Link>
+        );
+    }
+
+    renderDownloadableLink = (link) => {
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/ProductDownloadableLinks/ProductDownloadableLinks.component.js:98]: Please don't use arrow functions as class methods.
You cannot call the parent method when you are extending such functions.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

**Review by zans-laksa — APPROVED** [review]
Code review passed.

**Pull request merged**